controller/controlplane/controller.go

```diff
@@ -499,10 +499,27 @@ func (r *Reconciler) constructControlPlaneManagerConfigOptions(
 	}
 
 	if r.SecretLabelSelector != "" {
-		cfgOpts = append(cfgOpts, WithSecretLabelSelector(r.SecretLabelSelector))
+		cfgOpts = append(cfgOpts, WithSecretLabelSelectorMatchLabel(r.SecretLabelSelector, "true"))
 	}
+	if cp.Spec.ObjectFilters != nil && cp.Spec.ObjectFilters.Secrets != nil {
+		for k, v := range cp.Spec.ObjectFilters.Secrets.MatchLabels {
+			if k == r.SecretLabelSelector {
```

Comment:
REVIEW: How should we deal with conflicts between CP level and KO level label selectors?

- Return error in the reconciliation stage
- Ignore one of them and tell the behavior explicitly in docs
- Use admission webhooks (Not recommended, because we do not have a validating webhook for `ControlPlane` now)

Reply:
Use a status condition and set its status to `False` to signal this.